### ISSUE TITLE
Revise does not work in sheet, but works in the Juno REPL

### ISSUE THREAD

**Issue by xximranxx** [JunoLab/Juno.jl#107]
Trying to use Revise (@timholy) in the latest Juno (uber-juno 0.1.2)/Atom(1.23.3), julia 0.6.2

Revise works in a normal julia REPL and also in the Juno REPL but not when run in a sheet

A.jl:
```
module A
f()=1
end
```

Test.jl
```
push!(LOAD_PATH,pwd())
using Revise
using A
A.f() # gives 1
```

Change code in A.jl to e.g. `f()=2` and run `A.f()` in the Test.jl sheet again => no change
Doing the same in a REPL (julia or inside Atom) will return the new value of A.f()

If, after changing A.jl, we run any command in the Atom REPL, then running `A.f()` in the Test.jl sheet yields the new result

I had written this as an issue in Revise (https://github.com/timholy/Revise.jl/issues/78), but since it is working in the REPL's, thought it belongs here

**Comment by pfitzseb** [comment]
Right, this is unsurprising -- Revise.jl hooks into the REPL to call `revise` before evaluating user input (see [here](https://github.com/timholy/Revise.jl/blob/v0.6/src/Revise.jl#L774)), but Juno's inline evaluation doesn't use the REPL code at all. 

I suppose we could add a `isedfined(Main, :Revise) && Revise.revise()` call, but that seems a bit hackish. Could also develop an API that allows third-party Julia packages to register pre-eval and post-eval functions.

**Comment by xximranxx** [comment]
Would the option of registering pre-eval and post-eval functions not perhaps open up a "can of worms", in that installed packages might be, by mistake, hijacking completely independent evaluations, causing side-effects and such?

But I do think that solving this is important, as it seems to me that Revise will be a very important tool in larger project management

I do this the point that the first solution is a little hackish, but certainly has its worth

**Comment by RAbraham** [comment]
I got `Revise` working in my `runtests.jl` which is similar to evaluating a sheet(I think) by doing the following steps:

- Adding [this](https://github.com/timholy/Revise.jl#using-revise-by-default) to `~/.juliarc.jl`
- Open `runtests.jl` in Juno. 
- In `runtests.jl`, I have my `using MyModule` statement. Evaluate it. Now if you change the code in `MyModule.mymethod` and save, my test in `runtests.jl` picks up the change and fails/succeeds as expected. I tried different combinations and I think evaluating `using MyModule` statement in your sheet may be the key.

Hope that helps.
